sdk/src/main/java/io/opentelemetry/sdk/trace/Samplers.java

```diff
@@ -233,4 +244,33 @@ public boolean isSampled() {
       return Collections.emptyMap();
     }
   }
+
+  /** Probability-based sampling decision with a single attribute for the probability. */
+  @Immutable
+  @AutoValue
+  abstract static class ProbabilityDecision implements Decision {
+
+    ProbabilityDecision() {}
+
+    /**
+     * Creates sampling decision without attributes.
+     *
+     * @param decision sampling decision
+     * @param probability the probability that was used for the decision.
+     */
+    static ProbabilityDecision create(boolean decision, double probability) {
+      return new AutoValue_Samplers_ProbabilityDecision(
+          decision, singletonMap("samplingProbability", doubleAttributeValue(probability)));
+    }
+
+    @Override
+    public abstract boolean isSampled();
+
+    public abstract Map<String, AttributeValue> getAttributes();
+
+    @Override
+    public Map<String, AttributeValue> attributes() {
```

Comment:
To be more java we should use `getAttributes` in the Decision class I think.

Reply:
I agree. Should I make that part of this PR, or create an issue for it?

Reply:
up to you, happy to review that in this PR.

Reply:
done